props/websubmit/src/main.rs

```diff
@@ -177,6 +177,112 @@ fn run_sc_policy(ctx: Arc<Context>) -> Result<()> {
     })
 }
 
+/// If sensitive data is released, the release must be scoped, and all inputs to the scope must be safe.
+pub struct AuthDisclosureProp {
+    cx: Arc<PolicyContext>,
+}
+
+impl AuthDisclosureProp {
+    pub fn new(cx: Arc<PolicyContext>) -> Self {
+        AuthDisclosureProp { cx }
+    }
+
+    pub fn check(self) {
+        for c_id in self.cx.desc().controllers.keys() {
+            // All srcs that have no influencers
+            let roots = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|n| {
+                    self.cx
+                        .influencers(*n, paralegal_policy::EdgeType::DataAndControl)
+                        .next()
+                        .is_none()
+                })
+                .collect::<Vec<_>>();
+
+            let safe_scopes = self
+                .cx
+                // All nodes marked "safe"
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|n| self.cx.has_marker(marker!(safe_source), *n))
+                // And all nodes marked "safe_with_bless"
+                .chain(self.cx.all_nodes_for_ctrl(*c_id).filter(|node| {
+                    self.cx.has_marker(marker!(safe_source_with_bless), *node)
+                        && self
+                            .cx
+                            // That are influenced by a node marked "bless"
+                            .influencers(*node, paralegal_policy::EdgeType::DataAndControl)
+                            .any(|b| self.cx.has_marker(marker!(bless_safe_source), b))
+                }))
+                .collect::<Vec<_>>();
+            let sinks = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|n| self.cx.has_marker(marker!(sink), *n))
+                .collect::<Vec<_>>();
+            let mut sensitives = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(sensitive), *node));
+
+            let controller_valid = sensitives.all(|sens| {
+                sinks.iter().all(|sink| {
```

Comment:
Since we report the error inside we don't necessarily have to use `all` here, a `for` loop or `for_each` would also do. If you want to make sure it tells you the controller for which it fails you can use `Context::controller_contexts` to get a `Context` wrapper that also reports the controller on each `assert_error!`.

The usage is basically [this](https://brownsys.github.io/paralegal/libs/paralegal_policy/diagnostics/index.html#intended-workflow) just with getting the `ctx` from [here](https://brownsys.github.io/paralegal/libs/paralegal_policy/struct.Context.html#method.controller_contexts)

Reply:
done